### DISCUSSION TITLE
forecasting - get fitted values from evaluate function

### DISCUSSION THREAD

**Discussion by ArturDragunov** [alan-turing-institute/sktime#5446] [Q&A]
Question: In issue #888, a similar question was raised regarding how to obtain fitted values from a model. I'm currently looking for a similar solution with the _evaluate_ function. The [documentation](https://www.sktime.net/en/latest/api_reference/auto_generated/sktime.forecasting.model_evaluation.evaluate.html) indicates that users can extract _y_train_, _y_test_, and _y_pred_ values. Is there a way to also extract _y_fitted_ using this function? The workaround could be to refit the model with the best parameters and obtain _y_fitted_ separately from the evaluate function.
Also, is there a possibility to obtain fitted coefficient or other parameters of the fitted model from evaluate function?

**Comment by fkiraly** [comment]
Could you explain what you mean with `y_fitted`?

Regarding coeffs, there is currently no convenient way to get them from `evaluate`, that could be a feature request (appreciate an issue).

The advanced benchmarking framework should support that though, see https://github.com/sktime/sktime-workshop-pydata-prague-2023/blob/main/notebooks/05_evaluation_and_benchmarking.ipynb

FYI @hazrulakmal

**Reply by ArturDragunov** [reply]
Sure! "y_fitted" here refers to the in-sample predictions produced by the forecaster used during fitting inside the _evaluate()_. As extracting coefficients and y_fitted is a very possibly interrelated request, I created an issue #5452 with the description of both requests. I hope it's more clear now :) If not, please, let me know.
